src/Snackbar/Snackbar.js

```diff
@@ -0,0 +1,229 @@
+    if (this.props.open && nextProps.open &&
+        (nextProps.message !== this.props.message || nextProps.action !== this.props.action)) {
+      this.setState({
+        open: false,
+      });
+
+      clearTimeout(this.timerOneAtTheTimeId);
+      this.timerOneAtTheTimeId = setTimeout(() => {
+        this.setState({
+          message: nextProps.message,
+          action: nextProps.action,
+          open: true,
+        });
+      }, 400);
+    } else {
+      const open = nextProps.open;
+
+      this.setState({
+        open: open !== null ? open : this.state.open,
+        message: nextProps.message,
+        action: nextProps.action,
+      });
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.open !== this.state.open) {
+      if (this.state.open) {
+        this.setAutoHideTimer();
+        this.setTransitionTimer();
+      } else {
+        clearTimeout(this.timerAutoHideId);
+      }
+    }
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.timerAutoHideId);
+    clearTimeout(this.timerTransitionId);
+    clearTimeout(this.timerOneAtTheTimeId);
+  }
+
+  componentClickAway = () => {
+    if (this.timerTransitionId) {
+      // If transitioning, don't close the snackbar.
+      return;
+    }
+
+    if (this.props.open !== null && this.props.onRequestClose) {
+      this.props.onRequestClose('clickaway');
+    } else {
+      this.setState({ open: false });
+    }
+  };
+
+  // Timer that controls delay before snackbar auto hides
+  setAutoHideTimer() {
+    const autoHideDuration = this.props.autoHideDuration;
+
+    if (autoHideDuration > 0) {
+      clearTimeout(this.timerAutoHideId);
+      this.timerAutoHideId = setTimeout(() => {
+        if (this.props.open !== null && this.props.onRequestClose) {
+          this.props.onRequestClose('timeout');
+        } else {
+          this.setState({ open: false });
+        }
+      }, autoHideDuration);
+    }
+  }
+
+  /* Timer that controls delay before
+   click-away events are captured (based on when animation completes)
+  */
+  setTransitionTimer() {
+    this.timerTransitionId = setTimeout(() => {
+      this.timerTransitionId = undefined;
+    }, 400);
+  }
+  timerOneAtTheTimeId = undefined;
+  timerAutoHideId = undefined;
+  timerTransitionId = undefined;
+
+  render() {
+    const {
+      action,
+      message,
+      open,
+    } = this.state;
+
+    const {
+      autoHideDuration, // eslint-disable-line no-unused-vars
+      onRequestClose, // eslint-disable-line no-unused-vars
+      style,
+      ...other // eslint-disable-line no-unused-vars
+    } = this.props;
+
+    const dynamicStyle = {
+      visibility: open ? 'visible' : 'hidden',
+      transform: open ? 'translate(-50%, 0)' : 'translate(-50%, 48px)',
+    };
+
+    const classes = this.context.styleManager.render(styleSheet);
+    return (
+      <ClickAwayListener onClickAway={open ? this.componentClickAway : null}>
+        <div
```

Comment:
Snackbar content has its own className and styles there. I was trying to make a body that wraps the stateless snackbar. How would I inject styles like that in SnackbarContent  when that component already has a style defined there? I tried doing a couple things but it just made it uglier. Can you please elaborate?